vowpalwabbit/cb_adf.cc

```diff
@@ -490,6 +490,7 @@ base_learner* cb_adf_setup(options_i& options, vw& all)
   new_options
       .add(make_option("cb_adf", cb_adf_option)
                .keep()
+               .necessary()
```

Comment:
haven't implemented this, but I'm thinking in ways of elevating that if check into a something more concrete, since effectively this is how to turn on or off any reduction (which is helpful for pyatom). The 505 line change is optional, the boolean could make sense with a better name maybe?

Reply:
I really like this, I've been thinking we need some way to specify the main option that turns the reduction "on" in a more structured way

Reply:
opened this pr for this https://github.com/VowpalWabbit/vowpal_wabbit/pull/2531